internal/internal_event_handlers.go

```diff
@@ -690,6 +694,11 @@ func (wc *workflowEnvironmentImpl) MutableSideEffect(id string, f func() interfa
 	if result, ok := wc.mutableSideEffect[id]; ok {
 		encodedResult := newEncodedValue(result, wc.GetDataConverter())
 		if wc.isReplay {
+			// During replay, we only generate a command if there was a known marker
+			// recorded on the next task
+			if wc.mutableSideEffectsRecorded[id] {
```

Comment:
should we clear that flag here? we probably only want to generate the command once, even if there are 2 workflow.MutableSideEffect() calls with same id in this workflow task.

Reply:
I don't think it helps. This relates to a concern I brought up in Slack an in the description:

> Question: What if two mutable side effects of the same ID are run during the same WFT? We don't differentiate by command, only by user-defined marker ID.

The issue is that if you ran 20 of the exact same mutable side effect calls (say with timer creation in between each) in a single WFT, it would result in only one command and the other 19 are no-ops. If I cleared this each time, it'd result in 19 on replay if I understand correctly. There is no way to get replay of multiple mutable side effects w/ the same ID right since the only way you know whether a command should be generated is either via look ahead (we don't differentiate one mutable marker from another for the same ID) or actually running the callback.

I may be misunderstanding though.